### ISSUE TITLE
Crash on log4j-core jar

### ISSUE THREAD

**Issue by khitrin** [OpenJML/OpenJML#679]
OpenJML 0.8.41 crashes when trying import log4j-core.jar via classpath option.

```
[loading ZipFileIndexFileObject[log4j-core-2.11.0.jar(org/apache/logging/log4j/core/Core.class)]]
[loading ZipFileIndexFileObject[/usr/lib/jvm/java-1.8.0-openjdk-1.8.0.171-8.b10.el7_5.x86_64/lib/ct.sym(META-INF/sym/rt.jar/java/lang/Object.class)]]
Internal JML bug - please report.  BuildOpenJML-0.8.41-20190319
java.lang.NullPointerException
	at org.jmlspecs.openjml.JmlSpecs.setSpecsPath(JmlSpecs.java:439)
	at org.jmlspecs.openjml.JmlSpecs.setSpecsPath(JmlSpecs.java:415)
	at org.jmlspecs.openjml.JmlSpecs.initializeSpecsPath(JmlSpecs.java:273)
	at org.jmlspecs.openjml.JmlSpecs.getSpecsPath(JmlSpecs.java:389)
	at org.jmlspecs.openjml.JmlSpecs.findAnySpecFile(JmlSpecs.java:784)
	at org.jmlspecs.openjml.JmlCompiler.parseSpecs(JmlCompiler.java:218)
	at org.jmlspecs.openjml.JmlCompiler.completeBinaryEnterTodo(JmlCompiler.java:364)
	at org.jmlspecs.openjml.JmlCompiler.loadSpecsForBinary(JmlCompiler.java:335)
	at com.sun.tools.javac.comp.JmlResolve.loadClass(JmlResolve.java:172)
	at com.sun.tools.javac.comp.Resolve.findIdentInPackage(Resolve.java:2162)
	at com.sun.tools.javac.comp.Attr.selectSym(Attr.java:3403)
	at com.sun.tools.javac.comp.Attr.visitSelect(Attr.java:3290)
	at com.sun.tools.javac.comp.JmlAttr.visitSelect(JmlAttr.java:5290)
	at com.sun.tools.javac.tree.JCTree$JCFieldAccess.accept(JCTree.java:1897)
	at com.sun.tools.javac.comp.Attr.attribTree(Attr.java:577)
	at com.sun.tools.javac.comp.Attr.attribType(Attr.java:639)
	at com.sun.tools.javac.comp.Attr.attribType(Attr.java:632)
	at com.sun.tools.javac.comp.JmlAttr.attribType(JmlAttr.java:2419)
	at com.sun.tools.javac.comp.MemberEnter.attribImportType(MemberEnter.java:873)
	at com.sun.tools.javac.comp.MemberEnter.visitImport(MemberEnter.java:569)
	at com.sun.tools.javac.tree.JCTree$JCImport.accept(JCTree.java:571)
	at org.jmlspecs.openjml.JmlTree$JmlImport.accept(JmlTree.java:988)
	at com.sun.tools.javac.comp.MemberEnter.memberEnter(MemberEnter.java:438)
	at com.sun.tools.javac.comp.JmlMemberEnter.memberEnter(JmlMemberEnter.java:175)
	at com.sun.tools.javac.comp.MemberEnter.memberEnter(MemberEnter.java:450)
	at com.sun.tools.javac.comp.JmlMemberEnter.memberEnter(JmlMemberEnter.java:167)
	at com.sun.tools.javac.comp.JmlMemberEnter.importHelper(JmlMemberEnter.java:2153)
	at com.sun.tools.javac.comp.MemberEnter.visitTopLevel(MemberEnter.java:525)
	at com.sun.tools.javac.tree.JCTree$JCCompilationUnit.accept(JCTree.java:518)
	at org.jmlspecs.openjml.JmlTree$JmlCompilationUnit.accept(JmlTree.java:952)
	at com.sun.tools.javac.comp.MemberEnter.memberEnter(MemberEnter.java:438)
	at com.sun.tools.javac.comp.JmlMemberEnter.memberEnter(JmlMemberEnter.java:175)
	at com.sun.tools.javac.comp.Enter.complete(Enter.java:505)
	at com.sun.tools.javac.comp.Enter.main(Enter.java:471)
	at com.sun.tools.javac.comp.JmlEnter.main(JmlEnter.java:776)
	at com.sun.tools.javac.main.JavaCompiler.enterTrees(JavaCompiler.java:992)
	at com.sun.tools.javac.processing.JavacProcessingEnvironment$Round.enterTrees(JavacProcessingEnvironment.java:1015)
	at com.sun.tools.javac.processing.JavacProcessingEnvironment$Round.<init>(JavacProcessingEnvironment.java:902)
	at com.sun.tools.javac.processing.JavacProcessingEnvironment$Round.next(JavacProcessingEnvironment.java:921)
	at com.sun.tools.javac.processing.JavacProcessingEnvironment.doProcessing(JavacProcessingEnvironment.java:1187)
	at com.sun.tools.javac.main.JavaCompiler.processAnnotations(JavaCompiler.java:1180)
	at org.jmlspecs.openjml.JmlCompiler.processAnnotations(JmlCompiler.java:562)
	at com.sun.tools.javac.main.JavaCompiler.compile(JavaCompiler.java:863)
	at com.sun.tools.javac.main.Main.compile(Main.java:553)
	at com.sun.tools.javac.main.Main.compile(Main.java:410)
	at org.jmlspecs.openjml.Main.compile(Main.java:577)
	at com.sun.tools.javac.main.Main.compile(Main.java:399)
	at com.sun.tools.javac.main.Main.compile(Main.java:390)
	at org.jmlspecs.openjml.Main.execute(Main.java:413)
	at org.jmlspecs.openjml.Main.execute(Main.java:371)
	at org.jmlspecs.openjml.Main.execute(Main.java:358)
	at org.jmlspecs.openjml.Main.main(Main.java:330)
error: The compiler aborted with an uncaught exception: java.lang.IllegalStateException
java.lang.IllegalStateException
	at com.sun.tools.javac.util.Context.checkState(Context.java:225)
	at com.sun.tools.javac.util.Context.get(Context.java:146)
	at org.jmlspecs.openjml.Utils.instance(Utils.java:125)
	at org.jmlspecs.openjml.Main.execute(Main.java:415)
	at org.jmlspecs.openjml.Main.execute(Main.java:371)
	at org.jmlspecs.openjml.Main.execute(Main.java:358)
	at org.jmlspecs.openjml.Main.main(Main.java:330)
```

On macOS:
```
$ java -version
java version "1.8.0_201"
Java(TM) SE Runtime Environment (build 1.8.0_201-b09)
Java HotSpot(TM) 64-Bit Server VM (build 25.201-b09, mixed mode)

$ sw_vers 
ProductName:	Mac OS X
ProductVersion:	10.14.4
BuildVersion:	18E226
```

On Linux:
```
$ java -version
openjdk version "1.8.0_171"
OpenJDK Runtime Environment (build 1.8.0_171-b10)
OpenJDK 64-Bit Server VM (build 25.171-b10, mixed mode)
```

How to reproduce:

```
#!/bin/bash

wget https://github.com/OpenJML/OpenJML/releases/download/0.8.41/openjml-0.8.41-20190319.zip
wget http://central.maven.org/maven2/org/apache/logging/log4j/log4j-core/2.11.0/log4j-core-2.11.0.jar

echo "import org.apache.logging.log4j.core.Core; class Fail {}" > Fail.java

unzip openjml-0.8.41-20190319.zip

java -jar openjml.jar -verbose -cp log4j-core-2.11.0.jar Fail.java
```

**Comment by khitrin** [comment]
Just narrowed issue to single jar.

**Comment by davidcok** [comment]
Problem resolved in V 0.8.52 (and perhaps before that). Thanks for the clear repro
